### ISSUE TITLE
Varargs Issue

### ISSUE THREAD

**Issue by headjoe3** [roblox-ts/roblox-ts#221]
As of upgrading from 0.0.22 to 0.0.24, varargs are not transpiling correctly

```ts
    Run(...args: any[]): void {
        // print("Thread dispatch", this.name)
        // Connect event once
        let e: RBXScriptConnection
        e = this.eventObject.Event.Connect(() => {
            e.Disconnect()
            this.StartTimeoutCheck()
            this.func(...args)
            this.StopTimeoutCheck()
        })
        
        // Call and wait until thread yields or finishes executing
        this.eventObject.Fire()
    }
```
transpiles to
```lua
    Run = function(self, ...)
        local e;
        e = self.eventObject.Event:Connect(function()
            e:Disconnect();
            self:StartTimeoutCheck();
            self.func(unpack(args));
            self:StopTimeoutCheck();
        end);
        self.eventObject:Fire();
    end;
```

The line `self.func(unpack(args));` errors because "args" has never actually been defined.

For some reason this once-working functionality has been removed (git diff from before I updated)
![image](https://user-images.githubusercontent.com/4315566/52500305-e517a200-2b9a-11e9-9c27-6ae5bab43815.png)

**Comment by headjoe3** [comment]
Looks like this issue stems from an earlier decision to optimize varargs if they are always unpacked when they are needed....

The problem is that varargs always scope to the last-declared ancestor function, which actually breaks a lot of what Roblox-TS does, since things like ternary expressions, Array.push, etc. compile to
`(function() return value end)()` -type idioms.

Tracking every instance where the vararg scope changes in order to determine if this optimization will work seems unmanageable.
The logic went from this
![image](https://user-images.githubusercontent.com/4315566/52508137-c15f5680-2bb0-11e9-87f1-52368e09500e.png)
to this
![image](https://user-images.githubusercontent.com/4315566/52508152-d0de9f80-2bb0-11e9-8a5d-ba7b2543357a.png)

And extra logic would need to be added to that empty if clause if this were to be fixed. It's probably just best to revert to the old logic.

**Comment by evaera** [comment]
Fixed in ed0af373081ae9994b04fd7c6c612b53d6f1acf5
